org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultGradleWorkspaceManager.java

```diff
@@ -61,12 +62,18 @@ public GradleBuild getGradleBuild(BuildConfiguration buildConfig) {
 
     @Override
     public Optional<GradleBuild> getGradleBuild(IProject project) {
-        if (GradleProjectNature.isPresentOn(project)) {
-            BuildConfiguration buildConfig = CorePlugin.configurationManager().loadProjectConfiguration(project).getBuildConfiguration();
-            return Optional.<GradleBuild>of(new DefaultGradleBuild(buildConfig));
-        } else {
-            return Optional.absent();
-        }
+            try {
+                if (GradleProjectNature.isPresentOn(project)) {
+                    ProjectConfiguration projectConfiguration = CorePlugin.configurationManager().loadProjectConfiguration(project);
+                    BuildConfiguration buildConfig = projectConfiguration.getBuildConfiguration();
+                    return Optional.<GradleBuild>of(new DefaultGradleBuild(buildConfig));
+                } else {
+                    return Optional.absent();
+                }
+            } catch (RuntimeException e) {
+                CorePlugin.logger().debug("Cannot load configuration for project " + project.getName());
```

Comment:
I'm wondering if we should use Guava's `Optional` here. I can imagine that this might break yet another third party plugin that uses a more recent version of Guava.